src/realm/util/thread.hpp

```diff
@@ -435,28 +476,89 @@ inline Mutex::Mutex(process_shared_tag)
 
 inline Mutex::~Mutex() noexcept
 {
+#ifndef _WIN32
     int r = pthread_mutex_destroy(&m_impl);
```

Comment:
Nothing similar for windows? Aren't we going to leak a lot of resources, potentially?